client-spark/common/src/main/java/org/apache/spark/shuffle/reader/RssShuffleDataIterator.java

```diff
@@ -106,8 +108,11 @@ public boolean hasNext() {
       shuffleReadMetrics.incFetchWaitTime(fetchDuration);
       if (compressedData != null) {
         shuffleReadMetrics.incRemoteBytesRead(compressedData.limit() - compressedData.position());
```

Comment:
We should add some comments to explain why we need to do these `clean operation`.

Reply:
It seems better to extract a method and use reflection.
More information is as below
https://stackoverflow.com/questions/1854398/how-to-garbage-collect-a-direct-buffer-in-java